### DISCUSSION TITLE
How can I do spatial sounds?

### DISCUSSION THREAD

**Discussion by thefakeplace** [excaliburjs/Excalibur#2703] [Q&A]
I'd like to:
  - Play multiple sounds from different locations in the scene.
  - Have them mix properly based on location and distance relative to the camera.

Is this possible in Excalibur today?

**Comment by eonarheim** [comment]
@thefakeplace Good question!

We don't have anything built in for spatial audio yet, but this would be a great feature!

Looks like we should be able to implement something using a [panner node in the web audio api](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API/Web_audio_spatialization_basics)

As a workaround you may need to create new type based off `ex.WebAudioInstance` and change the `_createNewBufferSource()` to include a panner node.

```typescript
  private _createNewBufferSource() {
    this._instance = this._audioContext.createBufferSource();
    this._instance.buffer = this._src;
    this._instance.loop = this.loop;
    this._instance.playbackRate.value = this._playbackRate;
    this._instance.connect(this._volumeNode);
    this._volumeNode.connect(this._audioContext.destination);
  }
```

I'll have some time later to play around with this, but should be possible!

**Reply by eonarheim** [reply]
I'm thinking It might be nice to add some extension points to write custom audio code in Excalibur
